### PR TITLE
Added maxPacketParts method to the Network Card

### PR DESCRIPTION
**Pull request by brisingraerowing**
Ref #2294

### CODE REVIEW

**Comment by fnuecke** [comment]
I'd prefer this to go into the [device info](https://github.com/BrisingrAerowing/OpenComputers/blob/9d0b7438bcac338b799b281141e493cd177506bd/src/main/scala/li/cil/oc/server/component/NetworkCard.scala#L51) (hence the "todo remove" comment on the other, similar method). As for the attribute name, I suppose "width" could do? But feel free to use something more specific.

**Comment by skyem123** [comment]
It's worth noting that I feel like it should be on the network card component because it's like the MTU of real networks?

**Comment by skyem123** [comment]
It's worth noting that I feel like it should be on the network card component because it's like the MTU of real networks?

**Comment by fnuecke** [comment]
Not sure what your point is? The device info is still part of the network card component.

**Comment by skyem123** [comment]
I'm sorry... I don't really know what you're talking about with regards to "device info", as there doesn't seem to be anything on http://ocdoc.cil.li?

**Comment by Vexatos** [comment]
[This](https://github.com/BrisingrAerowing/OpenComputers/blob/9d0b7438bcac338b799b281141e493cd177506bd/src/main/scala/li/cil/oc/server/component/NetworkCard.scala#L46-L52) is the device info.

**Comment by brisingraerowing** [comment]
@fnuecke 

So how would that work? Would I add a new device attribute and set it to the value, similar to what is there?

What should that attribute be called?

**Comment by Vexatos** [comment]
Yes, exactly.

File: src/main/scala/li/cil/oc/server/component/NetworkCard.scala [hunk 1]
```diff
@@ -48,7 +48,8 @@ class NetworkCard(val host: EnvironmentHost) extends prefab.ManagedEnvironment w
     DeviceAttribute.Description -> "Ethernet controller",
     DeviceAttribute.Vendor -> Constants.DeviceInfo.DefaultVendor,
     DeviceAttribute.Product -> "42i520 (MPN-01)",
-    DeviceAttribute.Capacity -> Settings.get.maxNetworkPacketSize.toString
+    DeviceAttribute.Capacity -> Settings.get.maxNetworkPacketSize.toString,
+	DeviceAttribute.MaxParts -> Settings.get.maxNetworkPacketParts.toString
```
Comment (Vexatos) [inline src/main/scala/li/cil/oc/server/component/NetworkCard.scala]: Fix the indentation here.

File: src/main/java/li/cil/oc/api/driver/DeviceInfo.java [hunk 2]
```diff
@@ -69,7 +69,8 @@
         public static final String Size = "size"; // actual size of the device, e.g. actual usable space on a disk
         public static final String Clock = "clock"; // bus clock (in Hz) of the device, e.g. call speed(s) of a component
         public static final String Width = "width"; // address width of the device, in the broadest sense
-
+		public static final String MaxParts = "maxparts"; // Maximum number of parts for, e.g. a network packet.
```
Comment (Vexatos) [inline src/main/java/li/cil/oc/api/driver/DeviceInfo.java]: Fix the indentation here. I don't like the attribute name, but I can't think of anything other than @fnuecke's "width" either.

File: src/main/java/li/cil/oc/api/driver/DeviceInfo.java [hunk 3]
```diff
@@ -69,7 +69,8 @@
...
+
```
Comment (Vexatos) [inline src/main/java/li/cil/oc/api/driver/DeviceInfo.java]: Another formatting error. Remove the spaces.

**Review by Vexatos — CHANGES_REQUESTED** [review]

File: src/main/java/li/cil/oc/api/driver/DeviceInfo.java (hunk 3 above)
Comment (brisingraerowing) [reply src/main/java/li/cil/oc/api/driver/DeviceInfo.java]: Already done. Notepad++ Is rather stupid when it comes to Tabs vs Spaces.

File: src/main/scala/li/cil/oc/server/component/NetworkCard.scala [hunk 4]
```diff
@@ -48,8 +48,9 @@ class NetworkCard(val host: EnvironmentHost) extends prefab.ManagedEnvironment w
     DeviceAttribute.Description -> "Ethernet controller",
     DeviceAttribute.Vendor -> Constants.DeviceInfo.DefaultVendor,
     DeviceAttribute.Product -> "42i520 (MPN-01)",
-    DeviceAttribute.Capacity -> Settings.get.maxNetworkPacketSize.toString
-  )
+    DeviceAttribute.Capacity -> Settings.get.maxNetworkPacketSize.toString,
+    DeviceAttribute.MaxParts -> Settings.get.maxNetworkPacketParts.toString
+    )
```
Comment (Vexatos) [inline src/main/scala/li/cil/oc/server/component/NetworkCard.scala:53]: Still not formatted properly.

File: src/main/java/li/cil/oc/api/driver/DeviceInfo.java [hunk 5]
```diff
@@ -69,7 +69,7 @@
         public static final String Size = "size"; // actual size of the device, e.g. actual usable space on a disk
         public static final String Clock = "clock"; // bus clock (in Hz) of the device, e.g. call speed(s) of a component
         public static final String Width = "width"; // address width of the device, in the broadest sense
-
+        public static final String MaxParts = "maxparts"; // Maximum number of parts for, e.g. a network packet.
         private DeviceAttribute() {
```
Comment (Vexatos) [inline src/main/java/li/cil/oc/api/driver/DeviceInfo.java:73]: And an empty line here, please.

**Review by Vexatos — CHANGES_REQUESTED** [review]

**Pull request closed without merging**